lib/que/job.rb

```diff
@@ -101,6 +120,47 @@ def enqueue(*args)
       end
       ruby2_keywords(:enqueue) if respond_to?(:ruby2_keywords, true)
 
+      def bulk_enqueue(args_and_kwargs_array, job_options: {})
+        raise 'Unexpected bulk args format' if !args_and_kwargs_array.is_a?(Array) || !args_and_kwargs_array.all? { |a| a.is_a?(Hash) }
+
+        if job_options[:tags]
+          if job_options[:tags].length > MAXIMUM_TAGS_COUNT
+            raise Que::Error, "Can't enqueue a job with more than #{MAXIMUM_TAGS_COUNT} tags! (passed #{job_options[:tags].length})"
+          end
+
+          job_options[:tags].each do |tag|
+            if tag.length > MAXIMUM_TAG_LENGTH
+              raise Que::Error, "Can't enqueue a job with a tag longer than 100 characters! (\"#{tag}\")"
+            end
+          end
+        end
+
+        attrs = {
+          queue:    job_options[:queue]    || resolve_que_setting(:queue) || Que.default_queue,
+          priority: job_options[:priority] || resolve_que_setting(:priority),
+          run_at:   job_options[:run_at]   || resolve_que_setting(:run_at),
+          args:     Que.serialize_json(args_and_kwargs_array),
+          data:     job_options[:tags] ? Que.serialize_json(tags: job_options[:tags]) : "{}",
+          job_class: \
+            job_options[:job_class] || name ||
+              raise(Error, "Can't enqueue an anonymous subclass of Que::Job"),
+        }
+
+        if attrs[:run_at].nil? && resolve_que_setting(:run_synchronously)
+          attrs[:args] = Que.deserialize_json(attrs[:args])
+          attrs[:kwargs] = Que.deserialize_json(attrs[:kwargs])
```

Comment:
I don't think this line is relevant - `kwargs` is not used in this method, as everything is sent through to the query in `args`. Is synchronous mode even tested? =S

Reply:
Yeah, hadn’t got to updating this bit yet. I reckon it will be easy to support so we may as well.